src/lib/operators/projection.cpp

```diff
@@ -133,21 +133,23 @@ std::shared_ptr<const Table> Projection::_on_execute() {
                         ++chunk_offset;
                       }
                     });
+                    // As dictionary segments expect a BaseCompressedVector, we use a max-width (i.e., uint32_t)
+                    // FixedByteAlignedVector as we can directly forward the new attribute vector.
```

Comment:
DictionarySegments want a BaseCompressedVector, not an `std::vector<ValueId>` for the attribute vector. Luckily the latter can be wrapped into a FixedSizeByteAligned<uint32_t> without copying.

Reply:
```
// DictionarySegments take BaseCompressedVectors, not an std::vector<ValueId> for the attribute
// vector. But the latter can be wrapped into a FixedSizeByteAligned<uint32_t> without copying.
```

?